packages/core/src/consumed-thing.ts

```diff
@@ -121,19 +121,30 @@ export default class ConsumedThing extends TD.Thing implements WoT.ConsumedThing
         return form;
     }
 
-    ensureClientSecurity(client: ProtocolClient) {
-        // td-tools parser ensures this.security is an array
-        if (this.security && this.securityDefinitions && Array.isArray(this.security) && this.security.length > 0) {
-            console.debug("[core/consumed-thing]",`ConsumedThing '${this.title}' setting credentials for ${client}`);
-            let scs: Array<TD.SecurityScheme> = [];
-            for (let s of this.security) {
-                let ws = this.securityDefinitions[s + ""]; // String vs. string (fix wot-typescript-definitions?)
-                // also push nosec in case of proxy
-                if (ws) {
-                    scs.push(ws);
-                }
+
+    getSecuritySchemes(security: Array<String>): Array<TD.SecurityScheme> {
+        let scs: Array<TD.SecurityScheme> = [];
+        for (let s of security) {
+            let ws = this.securityDefinitions[s + ""]; // String vs. string (fix wot-typescript-definitions?)
```

Comment:
Is this comment still valid? 🤔

Reply:
What is strange is the following:
security on thing level is of type: `security: Array<String>`
security on form level is of type: `security?: Array<string>`
`String` vs. `string`

Reply:
I checked the new TD types and I don't see this problem. We should remember to remove the comment once we merge with 0.8.x